src/dispatcher.js

```diff
@@ -197,15 +197,11 @@ var dispatcher = {
   },
   leaveOut: function(event) {
     this.out(event);
-    if (!this.contains(event.target, event.relatedTarget)) {
-      this.leave(event);
-    }
+    this.propagate(event, this.leave.bind(this), false);
   },
   enterOver: function(event) {
     this.over(event);
-    if (!this.contains(event.target, event.relatedTarget)) {
-      this.enter(event);
-    }
+    this.propagate(event, this.enter.bind(this), true);
```

Comment:
Why not to pass the function itself to the `propagate` and do `fn.call(this)` there, instead of doing fn.bind, that is quite non-performant?